crates/typst/src/layout/inline/mod.rs

```diff
@@ -1446,3 +1489,56 @@ fn overhang(c: char) -> f64 {
         _ => 0.0,
     }
 }
+
+/// Whether the hyphen should repeat at the start of the next line.
+fn should_repeat_hyphen(pred_line: &Line) -> bool {
+    // If the predecessor line does not end with a Dash::HardHyphen, we shall not place a hyphen at
+    // the start of the next line.
+    if pred_line.dash != Some(Dash::HardHyphen) {
+        return false;
+    }
+
+    // If there's a trimmed out space, we needn't repeat the hyphen. That's the case of a text like
+    // "... kebab é a -melhor- comida que existe", where the hyphens are a kind of emphasis marker.
+    if pred_line.trimmed.end != pred_line.end {
+        return false;
+    }
+
+    // The hyphen should repeat only in the languages that requires that feature.
+    // For more information see the discussion at https://github.com/typst/typst/issues/3235
+    if let Some(Item::Text(shape)) = pred_line.last.as_ref() {
```

Comment:
WE could get rid of the indent here with `let .. else ..`.

Reply:
Nice hint! The code if cleaner that way!